hal_st/stm32fxxx/AnalogToDigitalPinStm.cpp

```diff
@@ -220,8 +220,10 @@ namespace hal
 
 #ifdef IS_ADC_SINGLE_DIFFERENTIAL
         result = HAL_ADCEx_Calibration_Start(&handle, ADC_SINGLE_ENDED);
-        assert(result == HAL_OK);
+#elif   defined(STM32WBA)
```

Comment:
This looks really asymmetrical. Moreover, the STM32G0 series also has a HAL_ADCEx_Calibration_Start with only a single parameter. Can you look for a better discriminating factor?

I also see that HAL_ADCEx_Calibration_Start is used in AdcDmaMultiChannelStm and AdcDmaStm. I assume this change needs to be applied there as well?

Reply:
It's very complex and difficult to oversee how all those discriminating factors work, and what is available. Can you come up with a discriminating factor that solves my issue of having no calibration on a WBA (but needing it), without influencing any other platforms/implementations that apparently don't need it.

Reply:
For me and my project that is fine.Don't I hit all cpu's (also none WBA), also ones that currently don't need this calibration? I have no way to cover and test all variations that I hit with this change, and cannot guarantee that another project is impacted by this change.

Reply:
About the other question: I have tested it on my hardware: when I add the HAL_ADCEx_Calibration_Start call to AdcDmaMultiChannelStm and AdcDmaStm my results are not worse or better than before. So from my perspective I don't mind adding it there as well. 
But again: how do I know if I don't break/influence other projects when I don't specifically tell I only want the change on platforms I can test/develop (WBA)?

Reply:
Can I get the blessing from someone that adding the HAL_ADCEx_Calibration_Start method to ADC's that are not classified as "ADC_SINGLE_ENDED" will not impact other projects and CPU's?

Reply:
@fabiangottstein Your suggestion won't work because not all CPU's supported by hal-st have a single parameter HAL_ADCEx_Calibration_Start() method in their libs.